doc/build/sysbuild/index.rst

```diff
@@ -648,3 +693,285 @@ each image that is part of a project. Alternatively, there are
+which should not be part of this group, then you can filter them out, like so:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     my_group
+     EXCLUDE hello_world
+   )
+
+Similarly, you can filter out entire groups of images:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     another_group
+     INCLUDE_FROM my_group
+     EXCLUDE_FROM other_group
+   )
+
+This example defines ``another_group``, which contains images that are unique to
+``my_group`` and do not overlap with ``other_group``. This illustrates a basic
+set difference operation for sysbuild groups.
+
+Exclusion takes precedence over inclusion when evaluating the contents of each
+group. For example, recall that the image ``my_sample`` has been added to both
+``my_group`` and ``other_group``, but it is filtered out from ``another_group``.
+Furthermore, trying to add that image back to the group by naïvely extending it:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     another_group
+     INCLUDE my_sample
+   )
+
+would not work, because the prior ``EXCLUDE_FROM`` directive still applies with
+higher priority. If you do need this to work, then you should consider defining
+yet another group:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     yet_another_group
+     INCLUDE_FROM another_group
+     INCLUDE my_sample
+   )
+
+In this way, ``yet_another_group`` can successfully include ``my_sample``,
+even though ``another_group`` excludes it. This is because group contents are
+evaluated in isolation, so by the time ``another_group`` is merged in, its own
+inclusions and exclusions will already have been squashed. The same goes for
+``my_group`` and ``other_group``, which need to be resolved like this first.
+
+Adding dependencies among sysbuild groups
+=========================================
+
+Just like for individual Zephyr applications, you can add configuration or
+flashing dependencies for groups of these applications.
+See :ref:`sysbuild_zephyr_application_dependencies` for more information.
+
+Here is an example of group-wise dependencies:
+
+.. code-block:: cmake
+
+   sysbuild_add_dependencies(GROUP CONFIGURE my_group group_a group_b)
+   sysbuild_add_dependencies(GROUP FLASH     my_group group_a group_b)
+
+This will ensure that every image in ``group_a`` and ``group_b`` will be
+configured and flashed before every image in ``my_group``. The ordering within
+the groups themselves is not determined here. Moreover, groups are not treated
+as units; once other image-wise and group-wise dependencies are factored in,
+the images belonging to one group are not expected to stick together in the
+sorted order.
+
+Note that ``my_group`` is allowed to overlap with ``group_a`` or ``group_b``.
+If two groups overlap and one group depends on the other, then the images that
+the two groups have in common will be placed in the middle of the sorted order,
+relative to the images unique to either group.
```

Comment:
When sharing properties of groups (which I understand is one desire with this PR) then it make sense to allow an image to belong (inherit) multiple groups.

But when it comes to dependency ordering, then I believe it makes it harder to understand the exact dependency handling compared to a restriction where an image may only belong to a single group.

We must therefore be extra careful in describing how dependencies are resolved, and just writing that two groups with images in common will be placed in the middle is not sufficient.

A way to describe it is that images which exists in both groups will not create dependency graphs between those images, they will however create dependency graphs to images only present in one of the groups.
Creating a dependency from group A to group B, and from B to A will not create a cyclic image dependency graph if A and B are identical.

In simple case, images i_1 and i_2, and groups g_1 and g_2:
```
g_1(i_1 i_2)
g_2(i_1 i_2)

sysbuild_add_dependencies(g_1 g_2)
sysbuild_add_dependencies(g_2 g_1)
sysbuild_groups_resolve(GROUPS g_1 g_2 IMAGES i_1 i_2)
```
Works, no cyclic image dependency.

But changing g_2 to only contain i_2 results in a loop:
```
g_1(i_1 i_2)
g_2(i_2)

sysbuild_add_dependencies(g_1 g_2)
sysbuild_add_dependencies(g_2 g_1)
sysbuild_groups_resolve(GROUPS g_1 g_2 IMAGES i_1 i_2)
```
Fails, i_1 <- i_2 (g_1 <- g_2), and i_2 <- i_1 (g_2 <- g_1)

One can still say the current text is acceptable on this, but let's add a third group g_3, and remove the group dep loop.
```
g_1(i_1 i_2)
g_2(    i_2)
g_3(i_1     i_3)

sysbuild_add_dependencies(g_1 g_2)
sysbuild_add_dependencies(g_1 g_3)
sysbuild_groups_resolve(GROUPS g_1 g_2 g_3 IMAGES i_1 i_2 i_3)
```
Fails, i_1 <- i_2 (g_1 <- g_2), and i_2 <- i_1 (g_1 <- g_3)

Now, in the latter case the text of `then the images that the two groups have in common will `be placed in the middle of the sorted order` is not clear imo.
```
sysbuild_add_dependencies(g_1 g_2):
    No dependency from i_2 to i_2, cause it exists in both groups.
    i_2 <- i_1, cause i_1 is unique in g_1.

sysbuild_add_dependencies(g_1 g_3):
    No dependency from i_1 to i_1, cause it exists in both groups.
    i_3 <- i_1, cause i_3 is unique in g_3.
    i_3 <- i_2, cause i_2 is unique in g_1 and i_3 is unique in g_3.
    i_1 <- i_2, cause i_2 is unique in g_1.
```

and now we can clearly see that because we have both `i_1 <-2` and `i_2 <- i1`, then there is a cyclic dependency.

I don't have a proposal for improved text, but if we want to allow an image to be present in multiple groups, then we need a much better description as to how image dependencies are generated in such cases.

Hopefully some of my description can help to come up with an improved text.